### PR TITLE
Di 20874 events list

### PR DESCRIPTION
**Pull request by roiaharon-pepperi**

### CODE REVIEW

File: client-side/src/app/events/events.component.ts [hunk 1]
```diff
@@ -11,13 +19,138 @@ export class EventsComponent implements OnInit {
...
+            this.dataSource = this.getDataSource();
+        })
     }
 
-    ngOnInit(): void {
+    getDataSource() {
+        return {
+            init: async(params:any) => {
+                this.events = await this.eventsService.getEvents();
+                return Promise.resolve({
```
Comment (Chasky-H) [inline client-side/src/app/events/events.component.ts:47]: no need for Promise.resolve when you marked the function as async and are using await.
You can just return the object. 
I think...

File: client-side/src/app/events/events.component.ts [hunk 2]
```diff
@@ -11,13 +19,138 @@ export class EventsComponent implements OnInit {
...
+                        this.showDeleteDialog(objs.rows[0]);
+                    }
+                })
+            }
+            return actions;
+        }
+    }
+
+    navigateToEventForm(name: string) {
+        this.router.navigate([name], {
```
Comment (Chasky-H) [inline client-side/src/app/events/events.component.ts:120]: so you are using a router in the end?
How will this work?

File: client-side/src/app/services/events-service.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        })
+    }
+
+    constructor(
+        public session:  PepSessionService,
+        private pepHttp: PepHttpService
+    ) {
+        const accessToken = this.session.getIdpToken();
+        this.parsedToken = jwt(accessToken);
+        this.papiBaseURL = this.parsedToken["pepperi.baseurl"];
```
Comment (Chasky-H) [inline client-side/src/app/services/events-service.ts:31]: Is this what everyone is doing??
Isn't there some service for this?

File: client-side/src/app/events/events.component.ts (hunk 1 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/events/events.component.ts:47]: I will try..

File: client-side/src/app/services/events-service.ts (hunk 3 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/services/events-service.ts:31]: as far as I know, there isn't. everyone is decoding to token to get the baseURL

File: client-side/src/app/events/events.component.ts (hunk 2 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/events/events.component.ts:120]: I'm not. this is a copy paste from the UDC, I will remove this when implementing the editing.

**Review by Chasky-H — APPROVED** [review]

**Pull request merged**